devops/jobs/RunRemoteAnsiblePlaybook.groovy

```diff
@@ -0,0 +1,111 @@
+
+class RunRemoteAnsiblePlaybook {
+    public static def job = { dslFactory, jobName, environment, deployment, extraVars ->
+        return dslFactory.job("${jobName}") {
+
+            // Common Jenkins Job features
+            wrappers common_wrappers
+
+            wrappers {
+              sshAgent(extraVars.get("SSH_AGENT_KEY", ''))
+            }
+
+            parameters {
+                stringParam('CONFIGURATION_REPO', extraVars.get('CONFIGURATION_REPO', 'https://github.com/edx/configuration.git'),
+                            'Git repo containing the analytics pipeline configuration automation.')
+                stringParam('CONFIGURATION_BRANCH', extraVars.get('CONFIGURATION_BRANCH', 'master'),
+                            'e.g. tagname or origin/branchname')
+                stringParam("SANDBOX_HOST",
+                            "staubina-ora2.sandbox.edx.org",
+                            "The host name and address of the Sandbox to be updated")
+                stringParam("ORA2_VERSION",
+                           "test2",
+                           "The host name and address of the Sandbox to be updated")
+                stringParam("NOTIFY_ON_FAILURE",
+                            "astaubin@edx.org",
+                            "Email to notify on failure")
+            }
+
+            def access_control = extraVars.get('ACCESS_CONTROL',[])
+            access_control.each { acl ->
+                common_read_permissions.each { perm ->
+                    authorization {
+                        permission(perm,acl)
+                    }
+                }
+            }
+
+            multiscm {
+                git {
+                    remote {
+                        url('$CONFIGURATION_REPO')
+                        branch('$CONFIGURATION_BRANCH')
+                    }
+                    extensions {
+                        cleanAfterCheckout()
+                        pruneBranches()
+                        relativeTargetDirectory('configuration')
+                    }
+                }
+            }
+
+
+            triggers {
+                if (extraVars.get('SCHEDULE')) {
+                    cron(extraVars['SCHEDULE'])
+                }
+            }
+            throttleConcurrentBuilds {
+                maxPerNode(1)
+            }
+
+            def ora2_version_str = extraVars.get("ORA2_VERSION", "test3")
+            // Parameters from the seed job that you don't override, they're just part of
+            // the run.  We need to load different ssh keys per env anyway.
+            environmentVariables {
+                env('PLAYBOOK', extraVars.get('PLAYBOOK'))
+                env('ANSIBLE_INVENTORY', extraVars.get('SANDBOX_HOST', ''))
+                env("REGION", extraVars.get('REGION','us-east-1'))
+                env("ANSIBLE_SSH_USER", extraVars.get('SSH_USER','ubuntu'))
+                env('ANSIBLE_EXTRA_VARS', "-e ora2_version=${ora2_version_str}")
```

Comment:
This is something you'll want to construct in the seed job, not in a generic task